### PR TITLE
Bug 2082763: Drop the OperatorHub CR instance

### PR DESCRIPTION
**Pull request by njhale**
- ~~Bump and vendor openshift/api to include a renaming of the OperatorHub
CRD yaml. This will preclude the CRD from cluster-config-operator's
CVO manifests and is part of an effort to make the marketplace-operator
optional via OpenShift's capabilities feature.~~ (openshift/api was bumped in another PR)
- Drop the OperatorHub CR instance

### CODE REVIEW

File: empty-resources/0000_05_config-operator_02_operatorhub.cr.yaml [hunk 1]
```diff
@@ -7,4 +7,5 @@ metadata:
     include.release.openshift.io/self-managed-high-availability: "true"
     include.release.openshift.io/single-node-developer: "true"
     release.openshift.io/create-only: "true"
+    capability.openshift.io/name: "marketplace"
```
Comment (njhale) [inline empty-resources/0000_05_config-operator_02_operatorhub.cr.yaml:10]: Should I actually be scrambling to move this to `/manifests` in `operator-marketplace`?

**Comment by njhale** [comment]
/hold

on operator-framework/operator-marketplace#468

**Comment by njhale** [comment]
/hold cancel

The PR this was blocked on has merged.

**Comment by tylerslaton** [comment]
Approach looks good and the vendor changes came through well. 

/lgtm

**Comment by awgreene** [comment]
I support the changes here, but we will need to ask QE to [update the OpenShift documentation](https://docs.openshift.com/container-platform/4.10/operators/understanding/olm-understanding-operatorhub.html#olm-operatorhub-arch-operatorhub_crd_olm-understanding-operatorhub) which states that it ships with an operatorhub resource named cluster. Instead, we'll need this section to ask that the user manually create the resource.

I was a bit worried that Marketplace won't behave as expected if the default operatorhub resource is not present, but [this code](https://github.com/operator-framework/operator-marketplace/blob/bd263c1cd7c6ab75b8e45d90f96c2974573f9ef0/cmd/manager/main.go#L254-L271) indicates that the default catalogSources would be created as expected.
/approve

**Comment by njhale** [comment]
> I support the changes here, but we will need to ask QE to [update the OpenShift documentation](https://docs.openshift.com/container-platform/4.10/operators/understanding/olm-understanding-operatorhub.html#olm-operatorhub-arch-operatorhub_crd_olm-understanding-operatorhub) which states that it ships with an operatorhub resource named cluster. Instead, we'll need this section to ask that the user manually create the resource.
> 
> I was a bit worried that Marketplace won't behave as expected if the default operatorhub resource is not present, but [this code](https://github.com/operator-framework/operator-marketplace/blob/bd263c1cd7c6ab75b8e45d90f96c2974573f9ef0/cmd/manager/main.go#L254-L271) indicates that the default catalogSources would be created as expected.
> 
> I don't have approval permissions in this repo, but I'll give my support!
> /approve

The operatorhub CRD is now included in marketplace as part of it's CVO manifests; i.e. it's all or nothing anyway.

**Comment by awgreene** [comment]
> > I support the changes here, but we will need to ask QE to [update the OpenShift documentation](https://docs.openshift.com/container-platform/4.10/operators/understanding/olm-understanding-operatorhub.html#olm-operatorhub-arch-operatorhub_crd_olm-understanding-operatorhub) which states that it ships with an operatorhub resource named cluster. Instead, we'll need this section to ask that the user manually create the resource.
> > I was a bit worried that Marketplace won't behave as expected if the default operatorhub resource is not present, but [this code](https://github.com/operator-framework/operator-marketplace/blob/bd263c1cd7c6ab75b8e45d90f96c2974573f9ef0/cmd/manager/main.go#L254-L271) indicates that the default catalogSources would be created as expected.
> > I don't have approval permissions in this repo, but I'll give my support!
> > /approve
> 
> The operatorhub CRD is now included in marketplace as part of it's CVO manifests; i.e. it's all or nothing anyway.

Ah I did not know that the cluster operatorhub CR was [moved](https://github.com/operator-framework/operator-marketplace/blob/bd263c1cd7c6ab75b8e45d90f96c2974573f9ef0/manifests/0000_03_marketplace-operator_02_operatorhub.cr.yaml) to the Marketplace repo, sgtm.

**Comment by njhale** [comment]
/hold

I believe this is blocked on operator-framework/operator-marketplace#472

TL;DR, It seems like we're depending on an old version of openshift/api in marketplace, which means that the CRD manifest is old and is hitting some relatively new logic in CVO that prevents manifests w/o annotations from being merged. I suspect that bumping and revendoring openshift/api in marketplace will resolve this issue.

**Comment by njhale** [comment]
looking for approval @bparees

**Comment by bparees** [comment]
@deads2k or @tkashem can you ptal?  this is finalizing moving the operatorhub CRD/CR manifest ownership away from config-operator and into the olm operator.

**Comment by wking** [comment]
/retitle Bug 2082763: Drop the OperatorHub CR instance

**Comment by jianzhangbjz** [comment]
Hi, I couldn't test it before merging via the `cluster-bot` since it doesn't support setting the `baselineCapabilitySet: None` flag at the install time, please move it on, thanks!

**Comment by bparees** [comment]
> Hi, I couldn't test it before merging via the cluster-bot since it doesn't support setting the baselineCapabilitySet: None flag at the install time, please move it on, thanks!

you can produce a payload from this PR using cluster-bot, and then do your own install w/ the None capset, right?

**Comment by perdasilva** [comment]
@kevinrizza Hey guys, what's missing for this to get merged? Manual testing by Jian? Could I ask that we please make this a priority. It would be great if we could get in asap, before Friday. Is there anything we can do to help expedite it?

**Comment by tkashem** [comment]
holding it until qe finishes their testing, feel free to cancel the hold once qe is happy

/hold

**Comment by bparees** [comment]
@jianzhangbjz are you able to test this by using cluster-bot to build a payload from this PR, and then install from that payload w/ the None capset?

**Comment by jianzhangbjz** [comment]
@bparees Hi, I'm PTO yesterday, will try it today, thanks!

**Comment by jianzhangbjz** [comment]
LGTM based on https://bugzilla.redhat.com/show_bug.cgi?id=2082763#c5
/lgtm

**Comment by jianzhangbjz** [comment]
@tkashem Could you help `/unhold` it? Thanks!
